irony-cdb.el

```diff
@@ -63,6 +63,20 @@ for files that it cannot handle."
   :type '(repeat function)
   :group 'irony-cdb)
 
+(defcustom irony-cdb-search-directory-list '("." "build")
```

Comment:
"." is the entry which covers the current behavior. Instead of special-casing it, check-subdirs loops over all defined directories. 

Removing build: I understood you agreed "build" was common enough. I think it is.
I disagree with you that customizing options is difficult. I also disagree that having extra
directories by default is something to be avoided .  Are you concerned about a scenario where:
1) a directory named `build` exists
2) it contains a json cdb.
3) **and** the user actually wants to use another cdb, which isn't
located in either the same directory or a parent directory.

....? and you believe that situation affects more users then those who use CMake and follow the `build/` convention which will now have irony working out of the box?